From 2d457ef64c08d916afaf82eccb0b69ab915acf45 Mon Sep 17 00:00:00 2001
From: Tom Stellard <tstellar@redhat.com>
Date: Thu, 3 Jul 2025 05:45:50 +0000
Subject: [PATCH] Fix build with clang-21

Error Message:

darktable-5.2.0/data/kernels/soften.cl:33:18: error: use of undeclared identifier 'read_imagef'
   33 |   float4 pixel = read_imagef(in, sampleri, (int2)(x, y));
      |                  ^~~~~~~~~~~

https://github.com/llvm/llvm-project/commit/c1aebd495be0e468044f716a3a0ff98fccccb2be
wrapped all the image function declarations in the __IMAGE_SUPPORT__
macro, so darktable needs to define this macro in order to use these
functions.
---
 data/kernels/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/kernels/CMakeLists.txt b/data/kernels/CMakeLists.txt
index 2e7fd0b309a7..48771063a3bc 100644
--- a/data/kernels/CMakeLists.txt
+++ b/data/kernels/CMakeLists.txt
@@ -15,7 +15,7 @@ macro (testcompile_opencl_kernel IN)
 
   add_custom_command(
     OUTPUT  ${TOUCH}
-    COMMAND ${CLANG_OPENCL_COMPILER} -cc1 -cl-std=CL1.2 -isystem ${CLANG_OPENCL_INCLUDE_DIR} -finclude-default-header -I${CMAKE_CURRENT_SOURCE_DIR} ${IN}
+    COMMAND ${CLANG_OPENCL_COMPILER} -cc1 -cl-std=CL1.2 -isystem ${CLANG_OPENCL_INCLUDE_DIR} -D__IMAGE_SUPPORT__=1 -finclude-default-header -I${CMAKE_CURRENT_SOURCE_DIR} ${IN}
     COMMAND ${CMAKE_COMMAND} -E touch ${TOUCH} # will be empty!
     DEPENDS ${IN}
     COMMENT "Test-compiling OpenCL program ${KERNAME}"
